sdk/resourcemanager/internal/testutil/recording.go

```diff
@@ -8,95 +8,49 @@ package testutil
 
 import (
 	"fmt"
-	"net/http"
-	"strconv"
 	"testing"
 
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
 	"github.com/Azure/azure-sdk-for-go/sdk/internal/recording"
 )
 
-const recordingRandomSeedVariableName = "recordingRandomSeed"
-
-var (
-	recordingSeed int64
-)
-
-type recordingPolicy struct {
-	options recording.RecordingOptions
-	t       *testing.T
-}
-
-// Host of the test proxy.
-func (r *recordingPolicy) Host() string {
-	if r.options.UseHTTPS {
-		return "localhost:5001"
-	}
-	return "localhost:5000"
-}
+// StartRecordingProxy starts the recording proxy with the path to store recording file.
+// It should be used in the module test preparation stage only once.
+// It will return a delegate function to stop recording proxy.
+func StartRecordingProxy(pathToPackage string) func() {
+	if recording.GetRecordMode() == recording.PlaybackMode || recording.GetRecordMode() == recording.RecordingMode {
+		proxy, err := recording.StartTestProxy(pathToPackage, nil)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to start recording proxy: %v", err))
+		}
 
-// Scheme of the test proxy.
-func (r *recordingPolicy) Scheme() string {
-	if r.options.UseHTTPS {
-		return "https"
-	}
-	return "http"
-}
+		// sanitizer for any uuid string, e.g., subscriptionID
+		err = recording.AddGeneralRegexSanitizer("00000000-0000-0000-0000-000000000000", `[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}`, proxy.Options)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to add uuid sanitizer: %v", err))
+		}
+		// consolidate resource group name for recording and playback
+		err = recording.AddGeneralRegexSanitizer("Sanitized", `go-sdk-test-\d+`, proxy.Options)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to add resource group name sanitizer: %v", err))
+		}
 
-// NewRecordingPolicy will create a recording policy which can be used in pipeline.
-// The policy will change the destination of the request to the proxy server and add required header for the recording test.
-func NewRecordingPolicy(t *testing.T, o *recording.RecordingOptions) policy.Policy {
-	if o == nil {
-		o = &recording.RecordingOptions{UseHTTPS: true}
+		return func() { StopRecordingProxy(proxy) }
```

Comment:
Consider inlining `StopRecordingProxy` here. I'm thinking there's no need to export it because it's useful only after starting the proxy, there's only one good way to do that, and a caller who followed the good pattern already has a function for stopping the proxy.